src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs

```diff
@@ -75,7 +77,7 @@ private void RevokeListeners()
 
 	protected override void OnApplyTemplate()
 	{
-		_canvas = GetTemplateChild(nameof(PART_Canvas)) as Canvas;
+		_canvas = GetTemplateChild(nameof(PART_Canvas)) as Grid;
```

Comment:
While it makes sense for me to use a `Grid` to ensure stretching, I would not enforce it's usage. I think that a `Panel` is enough for our internal usage.

Also you should consider to rename the `_canvas` backing field and the `UpdateCanvasSize` method ;)

Reply:
You can keep the `Grid` in the template, but reference to it as a `Panel` in the code (your are using only  `Children` and `Margin`), so user can change the panel in their app by just changing the template of the control.

Reply:
Now i get it. 
Thanks.
Changed.

Reply:
@dr1rrb 
Back to Grid, but kept the _panel, that is better than _owner